src/sidebar/audit-log.js

```diff
@@ -0,0 +1,129 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { CircularLoader } from '@dhis2/ui'
+import moment from 'moment'
+import PropTypes from 'prop-types'
+import React, { useEffect } from 'react'
+import styles from './audit-log.module.css'
+import LoadingError from './loading-error.js'
+import ToggleableUnit from './toggleable-unit.js'
+
+// TODO
+const query = {
+    auditLog: {
+        resource: 'auditLog',
+        id: ({ id }) => id,
+    },
+}
+
+// TODO: find better way to do this
+const renderMessage = ({ displayName, changeType, newValue, oldValue }) => {
+    switch (changeType) {
+        case 'UPDATE':
+            return i18n.t(
+                '{{displayName}} <span class="{{typeClass}}">updated</span> to {{newValue}} (was {{oldValue}})',
+                {
+                    displayName,
+                    typeClass: styles.entryType,
+                    newValue,
+                    oldValue,
+                }
+            )
+        case 'DELETE':
+            return i18n.t(
+                '{{displayName}} <span class="{{typeClass}}">deleted</span> (was {{oldValue}})',
+                {
+                    displayName,
+                    typeClass: styles.entryTypeDeleted,
+                    newValue,
+                    oldValue,
+                }
+            )
+        default:
+            return `${displayName} ${changeType}`
+    }
+}
+
+const Entry = ({ entry }) => {
+    const timeAgo = moment(entry.at).fromNow()
+
+    return (
+        <>
+            <span
+                className={styles.entryMessage}
+                dangerouslySetInnerHTML={{ __html: renderMessage(entry) }}
```

Comment:
If we cannot find a better solution for translations, it's probably worth a comment to explain why this is needed, and that this is safe since all values are actually sanitized - `i18next` sanitizes all values by default.

Reply:
Ah just realised that the default case is vulnerable to XSS. I'll fix that